From: Markus Koschany <apo@gambaru.de>
Date: Thu, 28 Feb 2013 17:45:50 +0100
Subject: fix segfault non-existent archive cmdl switch x

If using the command line switch -x, the user needs to specify a valid/existent
archive. If the name of the archive is mistyped, xarchiver will segfault. This
patch fixes the issue.

Bug: http://bugs.debian.org/701909
---
 src/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index a6a7f93..a6e05e3 100644
--- a/src/main.c
+++ b/src/main.c
@@ -141,7 +141,7 @@ int main (int argc, char **argv)
 		/* Switch -x */
 		if (extract_path != NULL)
 		{
-			if (argv[1] == NULL)
+			if (argv[1] == NULL || archive == NULL)
 			{
 				response = xa_show_message_dialog (NULL,GTK_DIALOG_MODAL,GTK_MESSAGE_ERROR,GTK_BUTTONS_OK,_("Can't extract files from the archive:"),_("You missed the archive name!\n"));
 				return -1;
